tracks/crystal/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,27 @@
+### Algorithm
+
+There are technically two ways to do this, but only one is useful.
+
+```crystal
+module Year
+  def self.leap?(year : Int32) : Bool
+    year.divisible_by?(4) && !year.divisible_by?(100) || year.divisible_by?(400)
```

Comment:
Depending on the order of operations, you might be missing parentheses around the test for 100 and 400.

Reply:
Parentheses are not necessary for the correct order of operations.

Reply:
Does that mean that if a year is NOT divisible by 4 (1023), it will NOT be checked against 400? As in:

- `false && !year.divisible_by?(100) || year.divisible_by?(400)`: shortcuts to false instantly
- and NOT: `false || year.divisible_by?(400)` -> `false || false`

Reply:
Crystal does use short-circuit evaluation, yes.
```crystal
1023.divisible_by?(4) && true
#=> false
1024.divisible_by?(4) && false || "a"
#=> "a"
```

Reply:
True enough. Adding that.